src/thor/trace_route_action.cc

```diff
@@ -36,6 +36,20 @@ struct MapMatch
   int edge_index = -1;
 };
 
+// TODO: this is a temp wrapper - it will be removed when MapMatcher::OfflineMatch is updated
+std::vector<std::pair<float, std::vector<meili::MatchResult>>> OfflineMatch(
+    const std::shared_ptr<meili::MapMatcher>& matcher,
+    std::vector<meili::Measurement>& sequence,
+    uint32_t best_paths) {
+  std::vector<meili::MatchResult> match_results = matcher->OfflineMatch(sequence, best_paths);
+  std::vector<std::pair<float, std::vector<meili::MatchResult>>> results;
+  float confidence_score = 1.0f / best_paths;
+  for (uint32_t k = 0; k < best_paths; ++k) {
```

Comment:
i wondered about this... for really short traces, its possible that there wont even be `k` results. like its possible that the number of combinations is `< k`. am i correct in understanding that this makes sure that there are k results? i think thats fine for now if so, but i hope that there is no where else in the code that we enforce, or assume that there will always be `k` results in total because in reality it can be `k` or less